src/components/Api.js

```diff
@@ -0,0 +1,88 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  // make a request and check the response
+  _request(url, options) {
+    return fetch(url, options).then(this._handleResponse);
+  }
+
+  // handle the response
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  _handleError(err) {
+    console.error("API Error:", err);
+  }
+
+  getUserInfo() {
+    return this._request(`${this._baseUrl}/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    }).catch(this._handleError);
```

Comment:
`    }).catch(this._handleError);
`

You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`   after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.